R/RE2/build_tarballs.jl

```diff
@@ -0,0 +1,52 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "RE2"
+tag = "2022-06-01"
+version = VersionNumber(replace(tag, "-" => "."))
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://github.com/google/re2/archive/refs/tags/$tag.tar.gz", "f89c61410a072e5cbcf8c27e3a778da7d6fd2f2b5b1445cd4f4508bee946ab0f"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+
+cd $WORKSPACE/srcdir/re2-*
+
+mkdir build_dir && cd build_dir
+
+CMAKE_FLAGS=(-DCMAKE_INSTALL_PREFIX=$prefix
+-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN}
+-DCMAKE_BUILD_TYPE=Release
+-DBUILD_SHARED_LIBS=ON)
+
+# To actually use shared_mutex
+if [[ ${target} == *mingw* ]]; then
+    CMAKE_FLAGS+=(-DCMAKE_CXX_FLAGS="-std=c++17")
+fi
```

Comment:
According to https://en.cppreference.com/w/cpp/thread/shared_mutex, `shared_mutex` was introduced in C++17.  If this function is used everywhere and not just for Windows, then this isn't Windows-specific.  Also, we should never set the standard in this way, the CMake file should set [`CMAKE_CXX_STANDARD`](https://cmake.org/cmake/help/latest/variable/CMAKE_CXX_STANDARD.html).  In other words, this is an upstream bug.

Reply:
I've reported the upstream bug and in the meantime here, I replace all occcurences of `cxx_std_11` in the CMakeLists.txt, which are used in multiple `target_compile_features` calls (`CMAKE_CXX_STANDARD` wasn't used).